controllers/replace_failed_process_groups_test.go

```diff
@@ -393,3 +1030,18 @@ func getRemovedProcessGroupIDs(cluster *fdbv1beta2.FoundationDBCluster) []fdbv1b
 	}
 	return results
 }
+
+func getPodByProcessGroupID(cluster *fdbv1beta2.FoundationDBCluster, processGroupID fdbv1beta2.ProcessGroupID) *corev1.Pod {
+	pods, err := clusterReconciler.PodLifecycleManager.GetPods(ctx.TODO(), clusterReconciler, cluster, internal.GetPodListOptions(cluster, "", "")...)
+	if err != nil {
```

Comment:
```suggestion
	processClass, id, err := internal.ParseProcessGroupID(processGroup)
	if err != nil {
		return err
	}
	pods, err := clusterReconciler.PodLifecycleManager.GetPods(ctx.TODO(), clusterReconciler, cluster, internal.GetPodListOptions(cluster, string(processClass), strconv.Itoa(id))...)
	if err != nil {
```

For a test case this is probably fine but this is quite expensive to query all Pods to get a single pod.

Reply:
the suggested change didn't work. but I like the idea and changed it accordingly.